src/NewCard/index.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+import * as React from "react";
+import styled from "styled-components";
+
+import Loading from "../Loading";
+import Wrapper from "./primitives/Wrapper";
+import { Provider as SectionProvider } from "./CardContext";
+import defaultTheme from "../defaultTheme";
+import getSpacingToken from "../common/getSpacingToken";
+import Header from "./Header";
+
+import type { Props } from "./index";
+
+const StyledCard = styled.div`
+  width: 100%;
+  box-sizing: border-box;
+  position: relative;
+  font-family: ${({ theme }) => theme.orbit.fontFamily};
+  margin-bottom: ${getSpacingToken};
+`;
+
+StyledCard.defaultProps = {
+  theme: defaultTheme,
+};
+
+const NewCard = ({
+  title,
+  icon,
+  actions,
+  description,
+  children,
+  loading,
+  spaceAfter,
+  dataA11ySection,
+}: Props) => {
+  const [expandedSections, setExpandedSections] = React.useState([]);
+
+  // handles array of expanded sections
+  const onToggle = (index: number) =>
+    expandedSections.indexOf(index) === -1
+      ? setExpandedSections([...expandedSections, index])
```

Comment:
I would rely on current/previous state with passing an arrow function in both cases:
```jsx
setExpandedSections(sections => [...sections, index])
```